internal/dnf/sql.go

```diff
@@ -2,21 +2,63 @@ package dnf
 
 import _ "embed" // embed sql
 
-// Return one column containing the repoid.
-// Takes six arguments, in order:
-//   - the "removed" enum
-//   - name
-//   - epoch
-//   - version
-//   - release
-//   - arch
-//
-//go:embed sql/repoid_for_package.sql
-var repoidForPackage string
+var (
+	// Return one column containing the repoid.
+	// Takes five arguments, in order:
+	//   - name
+	//   - epoch
+	//   - version
+	//   - release
+	//   - arch
+	//
+	//go:embed sql/dnf4_repoid_for_package.sql
+	dnf4RepoidForPackage string
+
+	// Return one column containing the repoids.
+	//
+	//go:embed sql/dnf4_all_repoids.sql
+	dnf4AllRepoids string
+)
+
+var (
+	// Return one column containing the repoid.
+	// Takes five arguments, in order:
+	//   - name
+	//   - epoch
+	//   - version
+	//   - release
+	//   - arch
+	//
+	//go:embed sql/dnf5_repoid_for_package.sql
+	dnf5RepoidForPackage string
 
-// Return one column containing the repoids.
-// Takes one argument:
-//   - the "removed" enum
+	// Return one column containing the repoids.
+	//
+	//go:embed sql/dnf5_all_repoids.sql
+	dnf5AllRepoids string
+)
+
+// Report the "version" value from the "config" table.
 //
-//go:embed sql/all_repoids.sql
-var allRepoids string
+//go:embed sql/get_db_version.sql
+var getDBVersion string
+
+// Queries holds the sql needed to examine the dnf database.
+type queries struct {
+	AllRepoids       string
+	RepoidForPackage string
+}
+
+// SqlForVersion associates a version string returned from executing
+// [getDBVersion] to the SQL queries for that version of the database schema.
+var sqlForVersion = map[string]queries{
```

Comment:
Annoyingly I think there are some cases where it's dnf4 and 1.1:

- DNF4: https://github.com/rpm-software-management/libdnf/blob/master/libdnf/transaction/sql/create_tables.sql#L108
- DNF5: https://github.com/rpm-software-management/dnf5/blob/main/libdnf5/transaction/db/sql/create_tables.sql#L157

Having said that I have no idea where 1.2 comes from :thinking:

Reply:
Yeah, I updated this with a check based on what tables are present.